internal/provider/aws_external_id_data_source.go

```diff
@@ -0,0 +1,114 @@
+// Code generated by Speakeasy (https://speakeasyapi.dev). DO NOT EDIT.
+
+package provider
+
+import (
+	"conductorone/internal/sdk"
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &AwsExternalIDDataSource{}
+var _ datasource.DataSourceWithConfigure = &AwsExternalIDDataSource{}
+
+func NewAwsExternalIDDataSource() datasource.DataSource {
+	return &AwsExternalIDDataSource{}
+}
+
+// AwsExternalIDDataSource is the data source implementation.
+type AwsExternalIDDataSource struct {
+	client *sdk.ConductoroneAPI
+}
+
+// AwsExternalIDDataSourceModel describes the data model.
+type AwsExternalIDDataSourceModel struct {
+	ExternalID types.String `tfsdk:"external_id"`
+}
+
+// Metadata returns the data source type name.
+func (r *AwsExternalIDDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_aws_external_id"
+}
+
+// Schema defines the schema for the data source.
+func (r *AwsExternalIDDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "AwsExternalID DataSource",
+
+		Attributes: map[string]schema.Attribute{
+			"external_id": schema.StringAttribute{
+				Computed:    true,
```

Comment:
is this also optional?

Reply:
No, it's read only. There are no optional fields for this data source.